docs/source/rest_api/error_codes.rst

```diff
@@ -0,0 +1,165 @@
+       fetch a single resource. Endpoints which return lists of resources will
+       simply return an empty list.
+   * - 500
+     - Internal Server Error
+     - Something is broken internally in the REST API or the validator. This may
+       be a bug, and if reproducable, should be reported.
+   * - 503
+     - Service Unavailable
+     - The REST API is unable to communicate with the validator. It may be down.
+       You should try your request later.
+
+
+JSON Response
+=============
+
+In the case of an error, rather than a *data* property, the JSON response will
+include a single *error* property with three values:
+
+   * *code* (integer) - a machine readable error code
+   * *title* (string) - a short headline for the error
+   * *message* (string) - a longer human readable description of what went wrong
+
+.. note::
+
+   While the title and message may change in the future, **the error code is
+   fixed**, and will always refer to this particular problem.
+
+
+Example JSON Response
+---------------------
+
+.. code-block:: json
+
+   {
+     "error": {
+       "code": 30,
+       "title": "Submitted Batches Invalid",
+       "message": "The submitted BatchList is invalid. It was poorly formed, or has an invalid signature."
+     }
+   }
+
+
+Error Codes and Descriptions
+============================
+
+.. list-table::
+   :widths: 4, 16, 60
+   :header-rows: 1
+
+   * - Code
+     - Title
+     - Description
+   * - 10
+     - Unknown Validator Error
+     - An unknown error occurred with the validator while processing the
+       request. This may be a bug, and if reproducable, should be reported.
+   * - 15
+     - Validator Not Ready
+     - The validator has no genesis block, and so cannot be queried. Wait for
+       genesis to be completed and resubmit. If you are running the validator,
+       ensure it was set up properly.
+   * - 17
+     - Validator Timed Out
+     - The request timed out while waiting for a response from the validator. It
+       may not be running, or have encountered an internal error. The request
+       may or may not have been processed.
+   * - 18
+     - Validator Disconnected
+     - The validator sent a disconnect signal while processing the response, and
+       is no longer available. Try your request again later.
+   * - 21
+     - Invalid Resource Header
+     - The validator sent back a resource with a header that could not be
+       decoded. There may be or a problem with the validator, or the data may
+       have been corrupted.
+   * - 27
+     - Unable to Fetch Statuses
+     - The validator should always return some status for every batch
+       requested. An unknown error caused statuses to be missing, and should be
+       reported.
+   * - 30
+     - Submitted Batches Invalid
+     - The submitted BatchList failed initial validation by the validator. It
+       may be poorly formed, or (most likely) has a bad signature.
```

Comment:
I would order this message such that the more likely item is first:

It may have a bad signature, or be poorly formed.